applications/nrf5340_audio/src/audio/audio_datapath.c

```diff
@@ -573,87 +573,94 @@ static void audio_datapath_i2s_blk_complete(uint32_t frame_start_ts, uint32_t *r
 	/********** I2S TX **********/
 	static uint8_t *tx_buf;
 
-	if (tx_buf_released != NULL) {
-		/* Double buffered index */
-		uint32_t next_out_blk_idx = NEXT_IDX(ctrl_blk.out.cons_blk_idx);
-
-		if (next_out_blk_idx != ctrl_blk.out.prod_blk_idx) {
-			/* Only increment if not in underrun condition */
-			ctrl_blk.out.cons_blk_idx = next_out_blk_idx;
-			if (underrun_condition) {
-				underrun_condition = false;
-				LOG_WRN("Data received, total underruns: %d",
-					ctrl_blk.out.total_blk_underruns);
-			}
+	if (IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL) || (CONFIG_AUDIO_DEV == HEADSET)) {
+		if (tx_buf_released != NULL) {
```

Comment:
Not a blocker: In general, when one gets up to four nested "ifs", it is time to consider refactoring.

Reply:
CI should catch any potential faults here, but have you tested CIS, unidir with I2S input?

Reply:
Tested it with CIS, unidir and I2S input and it worked fine!

Reply:
I2S TX in this contexts means "from nRF5340 to HW codec". This is only needed for headsets when it is unidir :)